types/src/generator/index.ts

```diff
@@ -144,39 +152,167 @@ function collectClasses(map: StructureMap): Set<string> {
   return classes;
 }
 
-export function generateDefinitions(root: StructureGroups): ts.Node[] {
-  const map = collectStructureMap(root);
-  const included = collectIncluded(map);
-  const classes = collectClasses(map);
+// Builds a map mapping structure names that are top-level nested types of
+// module structures to the names of those modules. Essentially, a map of which
+// modules export which types (e.g. "workerd::api::node::AsyncLocalStorage" =>
+// "node-internal:async_hooks"). We use this to make sure we don't include
+// duplicate definitions if an internal module references a type from another
+// internal module. In this case, we'll include the definition in the one that
+// exported it.
+function collectModuleTypeExports(
+  root: StructureGroups,
+  map: StructureMap
+): Map</* structureName */ string, /* moduleSpecifier */ string> {
+  const typeExports = new Map<string, string>();
+  root.getModules().forEach((module) => {
```

Comment:
nit: forEach is quite a bit slower in practice than a normal for loop.

Reply:
In this case `root.getModules()` is a `capnp-ts` API